internal/tools/cloudflare/dns_test.go

```diff
@@ -0,0 +1,536 @@
+		},
+		{
+			description:     "failure to get record ID",
+			customerDNSName: "customer.cloud.mattermost.com",
+			zoneID:          "RANDOMDIDFROMCLOUDFLARE",
+			setupName: func(zoneNameList []string, customerDNSName string) (zoneName string, found bool) {
+				return "cloud.mattermost.com", true
+			},
+			setupZoneID: func(zoneName string) (zoneID string, err error) {
+				return "RANDOMDIDFROMCLOUDFLARE", nil
+			},
+			setupRecordID: func(zoneID, customerDNSName string, logger logrus.FieldLogger) (recordID string, err error) {
+				return "", errors.New("failed to get record ID")
+			},
+			setupDeleteDNSRecord: func(ctx context.Context, zoneID, recordID string) error {
+				return nil
+			},
+			setupDNSRecord: func(ctx context.Context, zoneID string, rr cf.DNSRecord) ([]cf.DNSRecord, error) {
+				return []cf.DNSRecord{}, errors.New("failed to get DNS records")
+			},
+			awsZoneNameList: func() []string {
+				return []string{"cloud.mattermost.com"}
+			},
+			expected: errors.New("Failed to get record ID from Cloudflare for DNS: customer.cloud.mattermost.com: failed to get DNS Record ID from Cloudflare: failed to get DNS records"),
+		},
+		{
+			description:     "failure to delete DNS record",
+			customerDNSName: "customer.cloud.mattermost.com",
+			zoneID:          "RANDOMDIDFROMCLOUDFLARE",
+			setupName: func(zoneNameList []string, customerDNSName string) (zoneName string, found bool) {
+				return "cloud.mattermost.com", true
+			},
+			setupZoneID: func(zoneName string) (zoneID string, err error) {
+				return "RANDOMDIDFROMCLOUDFLARE", nil
+			},
+			setupRecordID: func(zoneID, customerDNSName string, logger logrus.FieldLogger) (recordID string, err error) {
+				return "RANDOMRECORDIDFROMCLOUDFLARE", nil
+			},
+			setupDeleteDNSRecord: func(ctx context.Context, zoneID, recordID string) error {
+				return errors.New("failed to delete DNS record")
+			},
+			setupDNSRecord: func(ctx context.Context, zoneID string, rr cf.DNSRecord) ([]cf.DNSRecord, error) {
+				return []cf.DNSRecord{}, errors.New("failed to get DNS records")
+			},
+			awsZoneNameList: func() []string {
+				return []string{"cloud.mattermost.com"}
+			},
+			expected: errors.New("Failed to get record ID from Cloudflare for DNS: customer.cloud.mattermost.com: failed to get DNS Record ID from Cloudflare: failed to get DNS records"),
+		},
+	}
+
+	for _, s := range samples {
+		t.Run(s.description, func(t *testing.T) {
+			mockCF.mockGetZoneName = s.setupName
+			mockCF.mockGetZoneID = s.setupZoneID
+			mockCF.mockGetRecordID = s.setupRecordID
+			mockCF.mockDeleteDNSRecord = s.setupDeleteDNSRecord
+			mockCF.mockDNSRecords = s.setupDNSRecord
+			mockAWS.mockGetPublicHostedZoneNames = s.awsZoneNameList
+			client := NewClientWithToken(mockCF, mockAWS)
+			err := client.DeleteDNSRecord(s.customerDNSName, logger)
+			fmt.Println(err)
+			if s.expected != nil {
+				assert.EqualError(t, s.expected, err.Error())
+			}
```

Comment:
Same here, can we use `assert.Equla`? Also we should probably remove `fmt.Println`.